app/Http/Controllers/PhpReboot/HomeController.php

```diff
@@ -14,6 +17,29 @@ class HomeController extends Controller
      */
     public function homePage()
     {
-        return view('PhpReboot.Home.homePage', ['menu' => 'home']);
+        $cacheTimeout = 1440;
+        
+        // Number of Magazines | Stores it in the cache
+        $issuesCount = Cache::remember('issueCount', $cacheTimeout, function(){
+           return Magazine::count(); 
+        });
+        
+        //Number of Articles
+        $articlesCount = Cache::remember('articlesCount', $cacheTimeout, function(){
+            return Article::count();
+        });
+        
+        //Last Article Date
+        $lastArticleDate = Cache::remember('lastArticleDate', $cacheTimeout, function(){
+            $lastArticle = Article::orderBy('published_at', 'Desc')->first();
+            return $lastArticle->published_at;
+        });
```

Comment:
Hi @aabi01 ,

Thanks for pull request. Please do not mind but this is not the right way of doing it. Let me explain:

**Thin controller**

Laravel follows MVC pattern. In MVC, we says we must have thin controller & fat model. Please note, here models are not exactly eloquent model but a model layer, which contains services, Repository & Model (Eloquent classes)

**Benefit of thin controller**

MVC or all other patterns basically says about **separation of concerns**. It exactly means that one class (or method) should be responsible for only one functionality. Do only one thing but do it completely and in best way. Same laws are given by **SOLID** principles where

- S - Single responsibility principle. Here our controller have multiple responsibilities (actually all)
- O - Open Close Principle, means our code is open for extension but close for modification. Here, if we need to change file cache to say memcache in future, we will need to update our controller. Thus my code is not closed for modification.
- L - Leskov Substitution Principle, means program should be replaceable with instances of their subtypes. Here, Cache is interface and cache drivers (file cache, memcache, Redis etc) are its instance (Concrete class). Here, we can not change concrete class without changing code and thus we are not following this principle.
- I - Interface segregation principle, not applicable here.
- D - Dependency Inversion Principle. Laravel provide it through IOC container (Service containers/Providers), which we are not using. Thus not following it.

Don't worry, it is still a very good attempt. I can't accept this PR but I'll take code from your repo and write it following all above principles. You can then relate my solution and your and confirm how we can do it in better way. I'm going to Diwali holidays from tomorrow so might need some time to implement it but I'll try to do it ASAP.

**Few links for more study**: (You can fine more using Google)

Thin controller: https://adamwathan.me/2013/10/02/putting-your-laravel-controllers-on-a-diet/
SOLID - https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)